ir/src/main/scala/org/scalajs/core/ir/Utils.scala

```diff
@@ -50,20 +52,44 @@ object Utils {
     /* Note that Java and JavaScript happen to use the same encoding for
      * Unicode, namely UTF-16, which means that 1 char from Java always equals
      * 1 char in JavaScript. */
+    val end = str.length
+    var i = 0
+    // scalastyle:off return
```

Comment:
The convention I've been using for `return`s and Scalastyle is to put the `:off` and `:on` directives at the beginning and end of the method, like this:
```scala
  def escapeJS(str: String): String = {
    // scalastyle:off return
    /* Note that ... */
    val end = str.length
    ...
    str
    // scalastyle:on return
  }
```
The advantage is that, when reading the method, the very first thing you see is: "Attention! There be `return`s in this method. Watch out for the control flow!" I find this visual warning interesting :-)